### ISSUE TITLE
Few JOOQ queries (REST API) cause Jetty's idleTimeout

### ISSUE THREAD

**Issue by marcomicera** [fasten-project/fasten#86]
## Describe the bug

This [endpoints table](https://github.com/fasten-project/fasten/wiki/API-endpoints-for-Maven-projects) includes the current status of every single endpoint.
The following endpoints use SQL queries that immediately return a result when executed from the PostgreSQL command line, but cause Jetty to timeout when executed with JOOQ:

- `/mvn/packages/{pkg}/versions`
- `/mvn/packages/{pkg}/{pkg_ver}/callgraph`
- `/mvn/packages/{pkg}/{pkg_ver}/callables`
- `/mvn/packages/{pkg}/{pkg_ver}/{fasten_uri}/metadata`
- `/mvn/packages/{pkg}/{pkg_ver}/edges`

## To Reproduce

1. Checkout to [`kbapi`](https://github.com/fasten-project/fasten/tree/kbapi).
2. Compile the core and start the REST server:
    ```bash
    mvn clean install -Dmaven.test.skip=true -f core/pom.xml && \
      PGPASSWORD=<KB_PWD> mvn clean install \
      -Dmaven.test.skip=true exec:java \
      -f analyzer/restapi-plugin/pom.xml \
      -Dexec.args='--port <REST_PORT> --database <KB_URL> --username <KB_USER>'
    ```
3. Perform a GET request to one of those endpoints, e.g.:
    ```bash
    curl http://localhost:<REST_PORT>/mvn/packages/com.lemoulinstudio.jnag:jnag-utils-jse-rds/versions
    ```

## Expected behavior
Those endpoints should return almost immediately.

## Logs

<details><summary>Another endpoint working fine</summary>

```
[qtp1997591039-21] DEBUG org.jooq.tools.LoggerListener - Executing query          : select "public"."packages"."package_name", "public"."package_versions"."version", "public"."package_versions"."metadata" from "public"."packages" join "public"."package_versions" on "public"."packages"."id" = "public"."package_versions"."package_id" where (1 = 1 and lower("public"."packages"."package_name") = lower(?) and lower("public"."package_versions"."version") = lower(?))
[qtp1997591039-21] DEBUG org.jooq.tools.LoggerListener - -> with bind values      : select "public"."packages"."package_name", "public"."package_versions"."version", "public"."package_versions"."metadata" from "public"."packages" join "public"."package_versions" on "public"."packages"."id" = "public"."package_versions"."package_id" where (1 = 1 and lower("public"."packages"."package_name") = lower('com.lemoulinstudio.jnag:jnag-utils-jse-rds') and lower("public"."package_versions"."version") = lower('1.2.1'))
[qtp1997591039-21] DEBUG org.jooq.tools.LoggerListener - Fetched result           : +------------------------------------------+-------+--------------------------------------------------+
[qtp1997591039-21] DEBUG org.jooq.tools.LoggerListener -                          : |package_name                              |version|metadata                                          |
[qtp1997591039-21] DEBUG org.jooq.tools.LoggerListener -                          : +------------------------------------------+-------+--------------------------------------------------+
[qtp1997591039-21] DEBUG org.jooq.tools.LoggerListener -                          : |com.lemoulinstudio.jnag:jnag-utils-jse-rds|1.2.1  |{"commitTag": "", "sourcesUrl": "http://172.16....|
[qtp1997591039-21] DEBUG org.jooq.tools.LoggerListener -                          : +------------------------------------------+-------+--------------------------------------------------+
[qtp1997591039-21] DEBUG org.jooq.tools.LoggerListener - Fetched row(s)           : 1
[qtp1997591039-21] DEBUG eu.fasten.core.data.metadatadb.MetadataDao - Total rows: 1
[qtp1997591039-21] DEBUG org.jboss.resteasy.resteasy_jaxrs.i18n - MessageBodyWriter: org.jboss.resteasy.spi.ResteasyProviderFactory$SortedKey
[qtp1997591039-21] DEBUG org.jboss.resteasy.resteasy_jaxrs.i18n - MessageBodyWriter: org.jboss.resteasy.plugins.providers.StringTextStar
[qtp1997591039-21] DEBUG org.jboss.resteasy.resteasy_jaxrs.i18n - MessageBodyWriter: org.jboss.resteasy.plugins.providers.StringTextStar
[qtp1997591039-21] DEBUG org.jboss.resteasy.resteasy_jaxrs.i18n - Interceptor Context: org.jboss.resteasy.core.interception.ServerWriterInterceptorContext,  Method : proceed
[qtp1997591039-21] DEBUG org.jboss.resteasy.resteasy_jaxrs.i18n - MessageBodyWriter: org.jboss.resteasy.spi.ResteasyProviderFactory$SortedKey
[qtp1997591039-21] DEBUG org.jboss.resteasy.resteasy_jaxrs.i18n - MessageBodyWriter: org.jboss.resteasy.plugins.providers.StringTextStar
[qtp1997591039-21] DEBUG org.eclipse.jetty.server.Server - handled=true async=false committed=false on HttpChannelOverHttp@4882f347{r=1,c=false,a=DISPATCHED,uri=//localhost:9090/mvn/packages/com.lemoulinstudio.jnag:jnag-utils-jse-rds/1.2.1/metadata,age=711}
[qtp1997591039-21] DEBUG org.eclipse.jetty.server.HttpChannelState - unhandle HttpChannelState@42a25c7a{s=DISPATCHED a=NOT_ASYNC i=true r=IDLE w=false}
[qtp1997591039-21] DEBUG org.eclipse.jetty.server.HttpChannel - HttpChannelOverHttp@4882f347{r=1,c=false,a=COMPLETING,uri=//localhost:9090/mvn/packages/com.lemoulinstudio.jnag:jnag-utils-jse-rds/1.2.1/metadata,age=712} action COMPLETE
[qtp1997591039-21] DEBUG org.eclipse.jetty.server.HttpChannel - sendResponse info=null content=DirectByteBuffer@3db1efbc[p=0,l=644,c=32768,r=644]={<<<{"fields":[{"sche...ies\\": []}}"]]}>>>\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00...\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00} complete=true committing=true callback=Blocker@642e9155{null}
[qtp1997591039-21] DEBUG org.eclipse.jetty.server.HttpChannel - COMMIT for /mvn/packages/com.lemoulinstudio.jnag:jnag-utils-jse-rds/1.2.1/metadata on HttpChannelOverHttp@4882f347{r=1,c=true,a=COMPLETING,uri=//localhost:9090/mvn/packages/com.lemoulinstudio.jnag:jnag-utils-jse-rds/1.2.1/metadata,age=712}
200 null HTTP/1.1
Date: Thu, 15 Oct 2020 11:26:53 GMT
Content-Type: application/json


[qtp1997591039-21] DEBUG org.eclipse.jetty.server.HttpConnection - org.eclipse.jetty.server.HttpConnection$SendCallback@38057c8a[PROCESSING][i=HTTP/1.1{s=200,h=2,cl=-1},cb=org.eclipse.jetty.server.HttpChannel$CommitCallback@13e215f0] generate: NEED_HEADER (null,[p=0,l=644,c=32768,r=644],true)@START
[qtp1997591039-21] DEBUG org.eclipse.jetty.http.HttpGenerator - generateHeaders HTTP/1.1{s=200,h=2,cl=-1} last=true content=DirectByteBuffer@3db1efbc[p=0,l=644,c=32768,r=644]={<<<{"fields":[{"sche...ies\\": []}}"]]}>>>\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00...\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00}
[qtp1997591039-21] DEBUG org.eclipse.jetty.http.HttpGenerator - Date: Thu, 15 Oct 2020 11:26:53 GMT
Content-Type: application/json


[qtp1997591039-21] DEBUG org.eclipse.jetty.http.HttpGenerator - CONTENT_LENGTH
[qtp1997591039-21] DEBUG org.eclipse.jetty.server.HttpConnection - org.eclipse.jetty.server.HttpConnection$SendCallback@38057c8a[PROCESSING][i=HTTP/1.1{s=200,h=2,cl=-1},cb=org.eclipse.jetty.server.HttpChannel$CommitCallback@13e215f0] generate: FLUSH ([p=0,l=141,c=8192,r=141],[p=0,l=644,c=32768,r=644],true)@COMPLETING
[qtp1997591039-21] DEBUG org.eclipse.jetty.io.WriteFlusher - write: WriteFlusher@340e468f{IDLE}->null [HeapByteBuffer@1fa07425[p=0,l=141,c=8192,r=141]={<<<HTTP/1.1 200 OK\r\n....v20180320)\r\n\r\n>>>trol: max-age=0\r\n...\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00},DirectByteBuffer@3db1efbc[p=0,l=644,c=32768,r=644]={<<<{"fields":[{"sche...ies\\": []}}"]]}>>>\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00...\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00}]
[qtp1997591039-21] DEBUG org.eclipse.jetty.io.WriteFlusher - update WriteFlusher@340e468f{WRITING}->null:IDLE-->WRITING
[qtp1997591039-21] DEBUG org.eclipse.jetty.io.ChannelEndPoint - flushed 785 SocketChannelEndPoint@6ea8184{/0:0:0:0:0:0:0:1:51647<->/0:0:0:0:0:0:0:1:9090,OPEN,fill=-,flush=W,to=743/30000}{io=0/0,kio=0,kro=1}->HttpConnection@35760168[p=HttpParser{s=END,0 of -1},g=HttpGenerator@60d0c2e4{s=COMPLETING}]=>HttpChannelOverHttp@4882f347{r=1,c=true,a=COMPLETING,uri=//localhost:9090/mvn/packages/com.lemoulinstudio.jnag:jnag-utils-jse-rds/1.2.1/metadata,age=734}
[qtp1997591039-21] DEBUG org.eclipse.jetty.io.WriteFlusher - Flushed=true written=785 remaining=0 WriteFlusher@340e468f{WRITING}->null
[qtp1997591039-21] DEBUG org.eclipse.jetty.io.WriteFlusher - update WriteFlusher@340e468f{IDLE}->null:WRITING-->IDLE
[qtp1997591039-21] DEBUG org.eclipse.jetty.server.HttpConnection - org.eclipse.jetty.server.HttpConnection$SendCallback@38057c8a[PROCESSING][i=HTTP/1.1{s=200,h=2,cl=-1},cb=org.eclipse.jetty.server.HttpChannel$CommitCallback@13e215f0] generate: DONE ([p=141,l=141,c=8192,r=0],[p=644,l=644,c=32768,r=0],true)@END
[qtp1997591039-21] DEBUG org.eclipse.jetty.server.HttpChannelState - onComplete HttpChannelState@42a25c7a{s=COMPLETING a=NOT_ASYNC i=false r=IDLE w=false}
[qtp1997591039-21] DEBUG org.eclipse.jetty.server.HttpChannel - COMPLETE for /mvn/packages/com.lemoulinstudio.jnag:jnag-utils-jse-rds/1.2.1/metadata written=644
[qtp1997591039-21] DEBUG org.eclipse.jetty.server.HttpChannelState - recycle HttpChannelState@42a25c7a{s=COMPLETED a=NOT_ASYNC i=false r=IDLE w=false}
[qtp1997591039-21] DEBUG org.eclipse.jetty.http.HttpParser - reset HttpParser{s=END,0 of -1}
[qtp1997591039-21] DEBUG org.eclipse.jetty.http.HttpParser - END --> START
[qtp1997591039-21] DEBUG org.eclipse.jetty.server.HttpChannel - HttpChannelOverHttp@4882f347{r=1,c=false,a=IDLE,uri=null,age=0} handle exit, result COMPLETE
[qtp1997591039-21] DEBUG org.eclipse.jetty.io.ChannelEndPoint - filled 0 SocketChannelEndPoint@6ea8184{/0:0:0:0:0:0:0:1:51647<->/0:0:0:0:0:0:0:1:9090,OPEN,fill=-,flush=-,to=7/30000}{io=0/0,kio=0,kro=1}->HttpConnection@35760168[p=HttpParser{s=START,0 of -1},g=HttpGenerator@60d0c2e4{s=START}]=>HttpChannelOverHttp@4882f347{r=1,c=false,a=IDLE,uri=null,age=0}
[qtp1997591039-21] DEBUG org.eclipse.jetty.io.ChannelEndPoint - filled 0 SocketChannelEndPoint@6ea8184{/0:0:0:0:0:0:0:1:51647<->/0:0:0:0:0:0:0:1:9090,OPEN,fill=-,flush=-,to=7/30000}{io=0/0,kio=0,kro=1}->HttpConnection@35760168[p=HttpParser{s=START,0 of -1},g=HttpGenerator@60d0c2e4{s=START}]=>HttpChannelOverHttp@4882f347{r=1,c=false,a=IDLE,uri=null,age=0}
[qtp1997591039-21] DEBUG org.eclipse.jetty.server.HttpConnection - HttpConnection@35760168[p=HttpParser{s=START,0 of -1},g=HttpGenerator@60d0c2e4{s=START}]=>HttpChannelOverHttp@4882f347{r=1,c=false,a=IDLE,uri=null,age=0}<-SocketChannelEndPoint@6ea8184{/0:0:0:0:0:0:0:1:51647<->/0:0:0:0:0:0:0:1:9090,OPEN,fill=-,flush=-,to=8/30000}{io=0/0,kio=0,kro=1}->HttpConnection@35760168[p=HttpParser{s=START,0 of -1},g=HttpGenerator@60d0c2e4{s=START}]=>HttpChannelOverHttp@4882f347{r=1,c=false,a=IDLE,uri=null,age=0} filled 0 HeapByteBuffer@1fa07425[p=0,l=0,c=8192,r=0]={<<<>>>HTTP/1.1 200 OK\r\n...\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00}
[qtp1997591039-21] DEBUG org.eclipse.jetty.server.HttpConnection - HttpConnection@35760168[p=HttpParser{s=START,0 of -1},g=HttpGenerator@60d0c2e4{s=START}]=>HttpChannelOverHttp@4882f347{r=1,c=false,a=IDLE,uri=null,age=0}<-SocketChannelEndPoint@6ea8184{/0:0:0:0:0:0:0:1:51647<->/0:0:0:0:0:0:0:1:9090,OPEN,fill=-,flush=-,to=8/30000}{io=0/0,kio=0,kro=1}->HttpConnection@35760168[p=HttpParser{s=START,0 of -1},g=HttpGenerator@60d0c2e4{s=START}]=>HttpChannelOverHttp@4882f347{r=1,c=false,a=IDLE,uri=null,age=0} parse HeapByteBuffer@1fa07425[p=0,l=0,c=8192,r=0]={<<<>>>HTTP/1.1 200 OK\r\n...\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00} {}
[qtp1997591039-21] DEBUG org.eclipse.jetty.http.HttpParser - parseNext s=START HeapByteBuffer@1fa07425[p=0,l=0,c=8192,r=0]={<<<>>>HTTP/1.1 200 OK\r\n...\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00}
[qtp1997591039-21] DEBUG org.eclipse.jetty.server.HttpConnection - HttpConnection@35760168[p=HttpParser{s=START,0 of -1},g=HttpGenerator@60d0c2e4{s=START}]=>HttpChannelOverHttp@4882f347{r=1,c=false,a=IDLE,uri=null,age=0}<-SocketChannelEndPoint@6ea8184{/0:0:0:0:0:0:0:1:51647<->/0:0:0:0:0:0:0:1:9090,OPEN,fill=-,flush=-,to=8/30000}{io=0/0,kio=0,kro=1}->HttpConnection@35760168[p=HttpParser{s=START,0 of -1},g=HttpGenerator@60d0c2e4{s=START}]=>HttpChannelOverHttp@4882f347{r=1,c=false,a=IDLE,uri=null,age=0} parsed false HttpParser{s=START,0 of -1}
[qtp1997591039-21] DEBUG org.eclipse.jetty.server.HttpConnection - releaseRequestBuffer HttpConnection@35760168[p=HttpParser{s=START,0 of -1},g=HttpGenerator@60d0c2e4{s=START}]=>HttpChannelOverHttp@4882f347{r=1,c=false,a=IDLE,uri=null,age=0}<-SocketChannelEndPoint@6ea8184{/0:0:0:0:0:0:0:1:51647<->/0:0:0:0:0:0:0:1:9090,OPEN,fill=-,flush=-,to=9/30000}{io=0/0,kio=0,kro=1}->HttpConnection@35760168[p=HttpParser{s=START,0 of -1},g=HttpGenerator@60d0c2e4{s=START}]=>HttpChannelOverHttp@4882f347{r=1,c=false,a=IDLE,uri=null,age=0}
[qtp1997591039-21] DEBUG org.eclipse.jetty.io.AbstractConnection - fillInterested HttpConnection@35760168[p=HttpParser{s=START,0 of -1},g=HttpGenerator@60d0c2e4{s=START}]=>HttpChannelOverHttp@4882f347{r=1,c=false,a=IDLE,uri=null,age=0}<-SocketChannelEndPoint@6ea8184{/0:0:0:0:0:0:0:1:51647<->/0:0:0:0:0:0:0:1:9090,OPEN,fill=-,flush=-,to=9/30000}{io=0/0,kio=0,kro=1}->HttpConnection@35760168[p=HttpParser{s=START,0 of -1},g=HttpGenerator@60d0c2e4{s=START}]=>HttpChannelOverHttp@4882f347{r=1,c=false,a=IDLE,uri=null,age=0}
[qtp1997591039-21] DEBUG org.eclipse.jetty.io.FillInterest - interested FillInterest@26c69379{AC.ReadCB@35760168{HttpConnection@35760168[p=HttpParser{s=START,0 of -1},g=HttpGenerator@60d0c2e4{s=START}]=>HttpChannelOverHttp@4882f347{r=1,c=false,a=IDLE,uri=null,age=0}<-SocketChannelEndPoint@6ea8184{/0:0:0:0:0:0:0:1:51647<->/0:0:0:0:0:0:0:1:9090,OPEN,fill=FI,flush=-,to=0/30000}{io=0/0,kio=0,kro=1}->HttpConnection@35760168[p=HttpParser{s=START,0 of -1},g=HttpGenerator@60d0c2e4{s=START}]=>HttpChannelOverHttp@4882f347{r=1,c=false,a=IDLE,uri=null,age=0}}}
[qtp1997591039-21] DEBUG org.eclipse.jetty.io.ChannelEndPoint - changeInterests p=false 0->1 for SocketChannelEndPoint@6ea8184{/0:0:0:0:0:0:0:1:51647<->/0:0:0:0:0:0:0:1:9090,OPEN,fill=FI,flush=-,to=0/30000}{io=0/1,kio=0,kro=1}->HttpConnection@35760168[p=HttpParser{s=START,0 of -1},g=HttpGenerator@60d0c2e4{s=START}]=>HttpChannelOverHttp@4882f347{r=1,c=false,a=IDLE,uri=null,age=0}
[qtp1997591039-21] DEBUG org.eclipse.jetty.io.ManagedSelector - Queued change org.eclipse.jetty.io.ChannelEndPoint$1@4067b2a7 on org.eclipse.jetty.io.ManagedSelector@73afe5e id=1 keys=1 selected=0 updates=0
[qtp1997591039-19] DEBUG org.eclipse.jetty.io.ManagedSelector - Selector sun.nio.ch.KQueueSelectorImpl@19a651e1 woken up from select, 0/1 selected
[qtp1997591039-19] DEBUG org.eclipse.jetty.io.ManagedSelector - Selector sun.nio.ch.KQueueSelectorImpl@19a651e1 processing 0 keys, 1 updates
[qtp1997591039-19] DEBUG org.eclipse.jetty.io.ManagedSelector - updateable 1
[qtp1997591039-19] DEBUG org.eclipse.jetty.io.ManagedSelector - update org.eclipse.jetty.io.ChannelEndPoint$1@4067b2a7
[qtp1997591039-19] DEBUG org.eclipse.jetty.io.ChannelEndPoint - Key interests updated 0 -> 1 on SocketChannelEndPoint@6ea8184{/0:0:0:0:0:0:0:1:51647<->/0:0:0:0:0:0:0:1:9090,OPEN,fill=FI,flush=-,to=1/30000}{io=1/1,kio=1,kro=1}->HttpConnection@35760168[p=HttpParser{s=START,0 of -1},g=HttpGenerator@60d0c2e4{s=START}]=>HttpChannelOverHttp@4882f347{r=1,c=false,a=IDLE,uri=null,age=0}
[qtp1997591039-19] DEBUG org.eclipse.jetty.io.ManagedSelector - updates 0
[qtp1997591039-19] DEBUG org.eclipse.jetty.io.ManagedSelector - Selector sun.nio.ch.KQueueSelectorImpl@19a651e1 waiting on select
[qtp1997591039-21] DEBUG org.eclipse.jetty.server.HttpConnection - HttpConnection@35760168[p=HttpParser{s=START,0 of -1},g=HttpGenerator@60d0c2e4{s=START}]=>HttpChannelOverHttp@4882f347{r=1,c=false,a=IDLE,uri=null,age=0}<-SocketChannelEndPoint@6ea8184{/0:0:0:0:0:0:0:1:51647<->/0:0:0:0:0:0:0:1:9090,OPEN,fill=FI,flush=-,to=1/30000}{io=1/1,kio=0,kro=1}->HttpConnection@35760168[p=HttpParser{s=START,0 of -1},g=HttpGenerator@60d0c2e4{s=START}]=>HttpChannelOverHttp@4882f347{r=1,c=false,a=IDLE,uri=null,age=0} onFillable exit HttpChannelState@42a25c7a{s=IDLE a=NOT_ASYNC i=true r=IDLE w=false} null
[qtp1997591039-21] DEBUG org.eclipse.jetty.util.thread.QueuedThreadPool - ran CEP:SocketChannelEndPoint@6ea8184{/0:0:0:0:0:0:0:1:51647<->/0:0:0:0:0:0:0:1:9090,OPEN,fill=FI,flush=-,to=2/30000}{io=1/1,kio=1,kro=1}->HttpConnection@35760168[p=HttpParser{s=START,0 of -1},g=HttpGenerator@60d0c2e4{s=START}]=>HttpChannelOverHttp@4882f347{r=1,c=false,a=IDLE,uri=null,age=0}:runFillable:BLOCKING
```

</details>

<details><summary>Jetty timeout</summary>

```
12:47:41.586 [qtp1703173670-21] DEBUG org.jooq.tools.LoggerListener - Executing query          : select "public"."edges"."source_id", "public"."edges"."target_id", "public"."edges"."receivers", "public"."edges"."metadata" from "public"."edges" join "public"."callables" on "public"."edges"."source_id" = "public"."callables"."id" join "public"."modules" on "public"."modules"."id" = "public"."callables"."module_id" join "public"."package_versions" on "public"."package_versions"."id" = "public"."modules"."package_version_id" join "public"."packages" on "public"."packages"."id" = "public"."package_versions"."package_id" where (1 = 1 and lower("public"."packages"."package_name") = lower(?) and lower("public"."package_versions"."version") = lower(?))
12:47:41.588 [qtp1703173670-21] DEBUG org.jooq.tools.LoggerListener - -> with bind values      : select "public"."edges"."source_id", "public"."edges"."target_id", "public"."edges"."receivers", "public"."edges"."metadata" from "public"."edges" join "public"."callables" on "public"."edges"."source_id" = "public"."callables"."id" join "public"."modules" on "public"."modules"."id" = "public"."callables"."module_id" join "public"."package_versions" on "public"."package_versions"."id" = "public"."modules"."package_version_id" join "public"."packages" on "public"."packages"."id" = "public"."package_versions"."package_id" where (1 = 1 and lower("public"."packages"."package_name") = lower('com.lemoulinstudio.jnag:jnag-utils-jse-rds') and lower("public"."package_versions"."version") = lower('1.2.1'))
12:48:11.158 [Scheduler-20081582] DEBUG org.eclipse.jetty.io.IdleTimeout - SocketChannelEndPoint@40c34170{/0:0:0:0:0:0:0:1:51318<->/0:0:0:0:0:0:0:1:9090,OPEN,fill=-,flush=-,to=29946/30000}{io=0/0,kio=0,kro=1}->HttpConnection@767f9a16[p=HttpParser{s=END,0 of -1},g=HttpGenerator@7fd061ea{s=START}]=>HttpChannelOverHttp@6f71a71e{r=1,c=false,a=DISPATCHED,uri=//localhost:9090/mvn/packages/com.lemoulinstudio.jnag:jnag-utils-jse-rds/1.2.1/edges,age=29935} idle timeout check, elapsed: 29946 ms, remaining: 54 ms
12:48:11.159 [Scheduler-20081582] DEBUG org.eclipse.jetty.io.IdleTimeout - SocketChannelEndPoint@5b57f627{/0:0:0:0:0:0:0:1:51319<->/0:0:0:0:0:0:0:1:9090,OPEN,fill=FI,flush=-,to=29957/30000}{io=1/1,kio=1,kro=0}->HttpConnection@2c52309f[p=HttpParser{s=START,0 of -1},g=HttpGenerator@6883fe6e{s=START}]=>HttpChannelOverHttp@43a0f6a5{r=0,c=false,a=IDLE,uri=null,age=0} idle timeout check, elapsed: 29957 ms, remaining: 43 ms
12:48:11.207 [Scheduler-20081582] DEBUG org.eclipse.jetty.io.IdleTimeout - SocketChannelEndPoint@5b57f627{/0:0:0:0:0:0:0:1:51319<->/0:0:0:0:0:0:0:1:9090,OPEN,fill=FI,flush=-,to=30005/30000}{io=1/1,kio=1,kro=0}->HttpConnection@2c52309f[p=HttpParser{s=START,0 of -1},g=HttpGenerator@6883fe6e{s=START}]=>HttpChannelOverHttp@43a0f6a5{r=0,c=false,a=IDLE,uri=null,age=0} idle timeout check, elapsed: 30005 ms, remaining: -5 ms
12:48:11.207 [Scheduler-20081582] DEBUG org.eclipse.jetty.io.IdleTimeout - SocketChannelEndPoint@5b57f627{/0:0:0:0:0:0:0:1:51319<->/0:0:0:0:0:0:0:1:9090,OPEN,fill=FI,flush=-,to=30005/30000}{io=1/1,kio=1,kro=0}->HttpConnection@2c52309f[p=HttpParser{s=START,0 of -1},g=HttpGenerator@6883fe6e{s=START}]=>HttpChannelOverHttp@43a0f6a5{r=0,c=false,a=IDLE,uri=null,age=0} idle timeout expired
12:48:11.211 [Scheduler-20081582] DEBUG org.eclipse.jetty.io.FillInterest - onFail FillInterest@7673e53c{AC.ReadCB@2c52309f{HttpConnection@2c52309f[p=HttpParser{s=START,0 of -1},g=HttpGenerator@6883fe6e{s=START}]=>HttpChannelOverHttp@43a0f6a5{r=0,c=false,a=IDLE,uri=null,age=0}<-SocketChannelEndPoint@5b57f627{/0:0:0:0:0:0:0:1:51319<->/0:0:0:0:0:0:0:1:9090,OPEN,fill=FI,flush=-,to=30006/30000}{io=1/1,kio=1,kro=0}->HttpConnection@2c52309f[p=HttpParser{s=START,0 of -1},g=HttpGenerator@6883fe6e{s=START}]=>HttpChannelOverHttp@43a0f6a5{r=0,c=false,a=IDLE,uri=null,age=0}}}
java.util.concurrent.TimeoutException: Idle timeout expired: 30005/30000 ms
	at org.eclipse.jetty.io.IdleTimeout.checkIdleTimeout(IdleTimeout.java:166)
	at org.eclipse.jetty.io.IdleTimeout$1.run(IdleTimeout.java:50)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
12:48:11.211 [Scheduler-20081582] DEBUG org.eclipse.jetty.http.HttpParser - close HttpParser{s=START,0 of -1}
12:48:11.211 [Scheduler-20081582] DEBUG org.eclipse.jetty.http.HttpParser - START --> CLOSE
12:48:11.212 [Scheduler-20081582] DEBUG org.eclipse.jetty.io.AbstractConnection - HttpConnection@2c52309f[p=HttpParser{s=CLOSE,0 of -1},g=HttpGenerator@6883fe6e{s=START}]=>HttpChannelOverHttp@43a0f6a5{r=0,c=false,a=IDLE,uri=null,age=0}<-SocketChannelEndPoint@5b57f627{/0:0:0:0:0:0:0:1:51319<->/0:0:0:0:0:0:0:1:9090,OPEN,fill=-,flush=-,to=30010/30000}{io=1/1,kio=1,kro=0}->HttpConnection@2c52309f[p=HttpParser{s=CLOSE,0 of -1},g=HttpGenerator@6883fe6e{s=START}]=>HttpChannelOverHttp@43a0f6a5{r=0,c=false,a=IDLE,uri=null,age=0} onFillInterestedFailed {}
12:48:11.212 [Scheduler-20081582] DEBUG org.eclipse.jetty.io.AbstractEndPoint - shutdownOutput SocketChannelEndPoint@5b57f627{/0:0:0:0:0:0:0:1:51319<->/0:0:0:0:0:0:0:1:9090,OPEN,fill=-,flush=-,to=30010/30000}{io=1/1,kio=1,kro=0}->HttpConnection@2c52309f[p=HttpParser{s=CLOSE,0 of -1},g=HttpGenerator@6883fe6e{s=START}]=>HttpChannelOverHttp@43a0f6a5{r=0,c=false,a=IDLE,uri=null,age=0}
12:48:11.213 [Scheduler-20081582] DEBUG org.eclipse.jetty.io.AbstractConnection - fillInterested HttpConnection@2c52309f[p=HttpParser{s=CLOSE,0 of -1},g=HttpGenerator@6883fe6e{s=START}]=>HttpChannelOverHttp@43a0f6a5{r=0,c=false,a=IDLE,uri=null,age=0}<-SocketChannelEndPoint@5b57f627{/0:0:0:0:0:0:0:1:51319<->/0:0:0:0:0:0:0:1:9090,OSHUT,fill=-,flush=-,to=30011/30000}{io=1/1,kio=1,kro=0}->HttpConnection@2c52309f[p=HttpParser{s=CLOSE,0 of -1},g=HttpGenerator@6883fe6e{s=START}]=>HttpChannelOverHttp@43a0f6a5{r=0,c=false,a=IDLE,uri=null,age=0}
12:48:11.229 [Scheduler-20081582] DEBUG org.eclipse.jetty.io.FillInterest - interested FillInterest@7673e53c{AC.ReadCB@2c52309f{HttpConnection@2c52309f[p=HttpParser{s=CLOSE,0 of -1},g=HttpGenerator@6883fe6e{s=START}]=>HttpChannelOverHttp@43a0f6a5{r=0,c=false,a=IDLE,uri=null,age=0}<-SocketChannelEndPoint@5b57f627{/0:0:0:0:0:0:0:1:51319<->/0:0:0:0:0:0:0:1:9090,OSHUT,fill=FI,flush=-,to=1/30000}{io=1/1,kio=1,kro=0}->HttpConnection@2c52309f[p=HttpParser{s=CLOSE,0 of -1},g=HttpGenerator@6883fe6e{s=START}]=>HttpChannelOverHttp@43a0f6a5{r=0,c=false,a=IDLE,uri=null,age=0}}}
12:48:11.229 [Scheduler-20081582] DEBUG org.eclipse.jetty.io.ChannelEndPoint - changeInterests p=false 1->1 for SocketChannelEndPoint@5b57f627{/0:0:0:0:0:0:0:1:51319<->/0:0:0:0:0:0:0:1:9090,OSHUT,fill=FI,flush=-,to=1/30000}{io=1/1,kio=1,kro=0}->HttpConnection@2c52309f[p=HttpParser{s=CLOSE,0 of -1},g=HttpGenerator@6883fe6e{s=START}]=>HttpChannelOverHttp@43a0f6a5{r=0,c=false,a=IDLE,uri=null,age=0}
12:48:11.229 [Scheduler-20081582] DEBUG org.eclipse.jetty.io.ManagedSelector - Queued change org.eclipse.jetty.io.ChannelEndPoint$1@1f40512 on org.eclipse.jetty.io.ManagedSelector@795c2808 id=2 keys=1 selected=0 updates=0
12:48:11.229 [qtp1703173670-18] DEBUG org.eclipse.jetty.io.ManagedSelector - Selector sun.nio.ch.KQueueSelectorImpl@2a08db6b woken up from select, 0/1 selected
12:48:11.229 [Scheduler-20081582] DEBUG org.eclipse.jetty.io.WriteFlusher - ignored: WriteFlusher@289da003{IDLE}->null
java.util.concurrent.TimeoutException: Idle timeout expired: 30005/30000 ms
	at org.eclipse.jetty.io.IdleTimeout.checkIdleTimeout(IdleTimeout.java:166)
	at org.eclipse.jetty.io.IdleTimeout$1.run(IdleTimeout.java:50)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
12:48:11.229 [qtp1703173670-18] DEBUG org.eclipse.jetty.io.ManagedSelector - Selector sun.nio.ch.KQueueSelectorImpl@2a08db6b processing 0 keys, 1 updates
12:48:11.229 [qtp1703173670-18] DEBUG org.eclipse.jetty.io.ManagedSelector - updateable 1
12:48:11.229 [qtp1703173670-18] DEBUG org.eclipse.jetty.io.ManagedSelector - update org.eclipse.jetty.io.ChannelEndPoint$1@1f40512
12:48:11.229 [Scheduler-20081582] DEBUG org.eclipse.jetty.io.AbstractEndPoint - Ignored idle endpoint SocketChannelEndPoint@5b57f627{/0:0:0:0:0:0:0:1:51319<->/0:0:0:0:0:0:0:1:9090,OSHUT,fill=FI,flush=-,to=1/30000}{io=1/1,kio=1,kro=0}->HttpConnection@2c52309f[p=HttpParser{s=CLOSE,0 of -1},g=HttpGenerator@6883fe6e{s=START}]=>HttpChannelOverHttp@43a0f6a5{r=0,c=false,a=IDLE,uri=null,age=0}
12:48:11.230 [qtp1703173670-18] DEBUG org.eclipse.jetty.io.ChannelEndPoint - Key interests updated 1 -> 1 on SocketChannelEndPoint@5b57f627{/0:0:0:0:0:0:0:1:51319<->/0:0:0:0:0:0:0:1:9090,OSHUT,fill=FI,flush=-,to=1/30000}{io=1/1,kio=1,kro=0}->HttpConnection@2c52309f[p=HttpParser{s=CLOSE,0 of -1},g=HttpGenerator@6883fe6e{s=START}]=>HttpChannelOverHttp@43a0f6a5{r=0,c=false,a=IDLE,uri=null,age=0}
12:48:11.230 [Scheduler-20081582] DEBUG org.eclipse.jetty.io.IdleTimeout - SocketChannelEndPoint@40c34170{/0:0:0:0:0:0:0:1:51318<->/0:0:0:0:0:0:0:1:9090,OPEN,fill=-,flush=-,to=30018/30000}{io=0/0,kio=0,kro=1}->HttpConnection@767f9a16[p=HttpParser{s=END,0 of -1},g=HttpGenerator@7fd061ea{s=START}]=>HttpChannelOverHttp@6f71a71e{r=1,c=false,a=DISPATCHED,uri=//localhost:9090/mvn/packages/com.lemoulinstudio.jnag:jnag-utils-jse-rds/1.2.1/edges,age=30007} idle timeout check, elapsed: 30017 ms, remaining: -17 ms
12:48:11.230 [qtp1703173670-18] DEBUG org.eclipse.jetty.io.ManagedSelector - updates 0
12:48:11.230 [qtp1703173670-18] DEBUG org.eclipse.jetty.io.ManagedSelector - Selector sun.nio.ch.KQueueSelectorImpl@2a08db6b waiting on select
12:48:11.230 [Scheduler-20081582] DEBUG org.eclipse.jetty.io.IdleTimeout - SocketChannelEndPoint@40c34170{/0:0:0:0:0:0:0:1:51318<->/0:0:0:0:0:0:0:1:9090,OPEN,fill=-,flush=-,to=30018/30000}{io=0/0,kio=0,kro=1}->HttpConnection@767f9a16[p=HttpParser{s=END,0 of -1},g=HttpGenerator@7fd061ea{s=START}]=>HttpChannelOverHttp@6f71a71e{r=1,c=false,a=DISPATCHED,uri=//localhost:9090/mvn/packages/com.lemoulinstudio.jnag:jnag-utils-jse-rds/1.2.1/edges,age=30007} idle timeout expired
12:48:11.230 [Scheduler-20081582] DEBUG org.eclipse.jetty.io.FillInterest - onFail FillInterest@44511e5a{null}
java.util.concurrent.TimeoutException: Idle timeout expired: 30017/30000 ms
	at org.eclipse.jetty.io.IdleTimeout.checkIdleTimeout(IdleTimeout.java:166)
	at org.eclipse.jetty.io.IdleTimeout$1.run(IdleTimeout.java:50)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
12:48:11.230 [Scheduler-20081582] DEBUG org.eclipse.jetty.io.WriteFlusher - ignored: WriteFlusher@6cc0c546{IDLE}->null
java.util.concurrent.TimeoutException: Idle timeout expired: 30017/30000 ms
	at org.eclipse.jetty.io.IdleTimeout.checkIdleTimeout(IdleTimeout.java:166)
	at org.eclipse.jetty.io.IdleTimeout$1.run(IdleTimeout.java:50)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
12:48:11.230 [Scheduler-20081582] DEBUG org.eclipse.jetty.io.AbstractEndPoint - Ignored idle endpoint SocketChannelEndPoint@40c34170{/0:0:0:0:0:0:0:1:51318<->/0:0:0:0:0:0:0:1:9090,OPEN,fill=-,flush=-,to=30018/30000}{io=0/0,kio=0,kro=1}->HttpConnection@767f9a16[p=HttpParser{s=END,0 of -1},g=HttpGenerator@7fd061ea{s=START}]=>HttpChannelOverHttp@6f71a71e{r=1,c=false,a=DISPATCHED,uri=//localhost:9090/mvn/packages/com.lemoulinstudio.jnag:jnag-utils-jse-rds/1.2.1/edges,age=30007}
```

</details>

## Environment
 - OS: macOS 10.15.6 (19G2021)
 - Java version: 11.0.2

**Comment by gousiosg** [comment]
This is because we don't have an index on `packages.name`, right?

**Comment by marcomicera** [comment]
> This is because we don't have an index on `packages.name`, right?

I cannot double-check it as the server seems to be down right now, but that wouldn't explain why those queries were executed immediately from the PostgreSQL cli.

There must be something going wrong with JOOQ, RESTeasy, or Jetty, I suppose...

**Comment by marcomicera** [comment]
> This is because we don't have an index on `packages.name`, right?

Ok now I see a multicolumn B-Tree index having `packages`.`package_name` as the leading (leftmost) column, hence being the most efficient one for that `WHERE` clause, so that's definitely not the issue.

**Comment by gousiosg** [comment]
What is the status with this issue? I also see similar exceptions in the server deployment.

**Comment by marcomicera** [comment]
> What is the status with this issue? I also see similar exceptions in the server deployment.

Since I didn't manage to enable CDI in RESTeasy (needed for testing #87), I might switch to something else like Spring, hence we would not be having those problems anymore.

Right now I have to postpone this as I have to work on another deliverable, but @romatallinn will assist me next week I guess.

**Comment by marcomicera** [comment]
Solved in #123.
